src/plugins/kdb/ldap/libkdb_ldap/ldap_misc.c

```diff
@@ -1330,6 +1330,43 @@ free_princ_ent_contents(osa_princ_ent_t princ_ent)
     princ_ent->old_key_len = 0;
 }
 
+/* Get any auth indicator values from LDAP and update the "require_auth"
+ * string */
+static krb5_error_code
+get_ldap_auth_ind(krb5_context context, LDAP *ld, LDAPMessage *ldap_ent,
+                  krb5_db_entry *entry)
+{
+    krb5_error_code ret;
+    int i;
+    char **auth_inds = NULL;
+    struct k5buf buf = EMPTY_K5BUF;
+
+    auth_inds = ldap_get_values(ld, ldap_ent, "krbPrincipalAuthInd");
+    if (auth_inds == NULL)
+        return 0;
+
+    k5_buf_init_dynamic(&buf);
+
+    /* space seperated list of indicators */
+    for (i = 0; auth_inds[i] != NULL; i++) {
+        k5_buf_add(&buf, auth_inds[i]);
+        if (auth_inds[i + 1] != NULL)
+            k5_buf_add(&buf, " ");
+    }
+
+    ret = k5_buf_status(&buf);
+    if (ret != 0)
+        goto cleanup;
+
+    if (buf.data != NULL)
+        krb5_dbe_set_string(context, entry, KRB5_KDB_SK_REQUIRE_AUTH,
+                            (char *)buf.data);
```

Comment:
We require braces around any flow control statement body which doesn't fit on one line, even if it's only one statement.  buf.data does not need a cast, as C supports implicit casts from void *.  Finally, you are discarding the return value of krb5_dbe_set_string() here; it should be assigned to ret.